contracts/minter-suite/Minters/MinterDAExpRefund/MinterDAExpRefundV0.sol

```diff
@@ -0,0 +1,997 @@
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to msg.sender.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefund(uint256 _projectId) external {
+        claimRefundTo(payable(msg.sender), _projectId);
+    }
+
+    /**
+     * @notice Refunds the sender's payment above current amount due for
+     * project `_projectId`. The current amount due is the the price paid for
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to address `_to`.
+     * @param _to Address to send refund to.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefundTo(address payable _to, uint256 _projectId)
+        public
+        nonReentrant
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Receipt storage receipt = receipts[msg.sender][_projectId];
+        // CHECKS
+        // input validation
+        require(_to != address(0), "No refund to the zero address");
+        // require that a user has purchased at least one token on this project
+        require(receipt.numPurchased > 0, "No purchases made by this address");
+        // get the latestPurchasePrice, which returns the sellout price if the
+        // auction sold out before reaching base price, or returns the base
+        // price if auction has reached base price and artist has withdrawn
+        // revenues.
+        // @dev if user is elligible for a refund, they have purchased a token,
+        // therefore we are guaranteed to not have a populated
+        // latestPurchasePrice
+        uint256 netTokenPrice = _projectConfig.latestPurchasePrice;
+
+        // EFFECTS
+        // calculate the refund amount
+        // implicit overflow/underflow checks in solidity ^0.8
+        uint256 amountDue = receipt.numPurchased * netTokenPrice;
+        uint256 refund = receipt.netPaid - amountDue;
+        // reduce the netPaid (in storage) to value after refund deducted
+        receipt.netPaid = amountDue;
+        // emit event indicating new receipt state
+        emit ReceiptUpdated(
+            msg.sender,
+            _projectId,
+            receipt.numPurchased,
+            receipt.netPaid
+        );
+
```

Comment:
Could it help us either to emit a ReceiptClaimed event here, or change ReceiptUpdated to add a `totalClaimed` maybe?

I'm thinking the website will need to send a list of `_projectIDs` to `claimRefundsTo()`, and that it will need to retrieve some map of {projectID: numRefundableInvocations, ...} per user. This is something we could maybe index on the subgraph, but I think we'd need an event trigger to track & delete `numRefundableInvocations` after a claim's been made.

Reply:
If all we know is a user's netPaid and numPurchased, as well as the project's latestPurchasePrice on the minter, we can index a stable calculated field of "excess claimable funds" simply equal to `netPaid - (numPurchased * latestPurchasePrice)`. Happy to chat about this more, but this lets us be very efficient in our indexing and querying of "which projects does this user address have claimable funds > 0 ETH on?"

Reply:
Awesome, yeah "Claimable funds" will def be helpful but I'm not sure how the front-end will efficiently be able to grab a user's list of all Project ID's for which a refund claim is available. 

Judging by the new Receipt entity we're adding on the subgraph, we'll be able to loop all the receipts for any given user and get their project IDs. So it's outside the scope of our smart contract but do you think this is something we'd maybe want a computed field for, too?

Reply:
Potentially, but I think the design document currently recommends just filtering for receipts with excess claimable funds greater than zero in the frontend graphql query.

I'm going to resolve this here for this PR's sake, but definitely chat with @Jbern16 about this (as he will be implementing the frontend portion of this minter) 🙏